### PR TITLE
Run coverage on travis

### PR DESCRIPTION
**Pull request by stsewd**
Our coverage is good, and I would love to maintain it like that. The coverage report is 0% on management commands (not sure if we care to have tests for that)

What change? Almost nothing, devs can still run normal tests, as usual. The only thing is that Travis will fail if the coverage goes down 79% (our current coverage).

We can increase the coverage later, but keeping it >=79% is a good first step.

If you want to check the coverage locally, you'll need to run tox like this `tox -e py36,coverage`

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -54,11 +54,8 @@ commands =
...
 description = run test suite with code coverage for the application with {basepython}
-deps =
-    -r{toxinidir}/requirements/testing.txt
-    pytest-cov
+deps = coverage
 whitelist_externals = echo
 commands =
-    py.test --disable-pytest-warnings \
-        --cov-report=term --cov-report=html --cov-config {toxinidir}/.coveragerc --cov=.  {posargs}
+    coverage report --show-missing --fail-under=79
```
Comment (stsewd) [inline tox.ini]: 79 is our current percent

Comment (stsewd) [reply tox.ini]: Well, looks like it is on 80% on travis, this is for the search tests I guess

**Comment by stsewd** [comment]
> why there are test files saying that some lines are missing? aren't those tests being executed? why? are they marked as @skip

Those aren't being executed at the moment but are unskiped in other PRs.

**Comment by stsewd** [comment]
> I think it will be better if we execute the coverage command as a separate job under travis. That way, we can immediately know if it fails because of the coverage or because of a test.

Not sure about that, It will slow down the CI even more :/

**Comment by stsewd** [comment]
Also, we could easily see the tox report to see which command fail, tests or coverage.

File: tox.ini [hunk 2]
```diff
@@ -54,11 +54,8 @@ commands =
...
 description = run test suite with code coverage for the application with {basepython}
-deps =
-    -r{toxinidir}/requirements/testing.txt
-    pytest-cov
+deps = coverage
 whitelist_externals = echo
 commands =
-    py.test --disable-pytest-warnings \
-        --cov-report=term --cov-report=html --cov-config {toxinidir}/.coveragerc --cov=.  {posargs}
+    coverage report --show-missing --fail-under=76
```
Comment (stsewd) [inline tox.ini]: And with the search tests we are at 78%, should I increase the coverage?

**Comment by stsewd** [comment]
Yeah, integrating with codecov looks good, and it uses coverage so we can integrate that easily here. I'll update this PR to integrate with codecov

**Comment by humitos** [comment]
`codecov` is good, but it's too verbose. Is this comment added on every commit to the PR? If so, I think we will end up with many comments from this bot on PRs that will be impossible to follow.

I think we should tweak codecov to behave in a way that doesn't make hard to read all the comments of the PR. Maybe only the line 

    Merging #5549 into 1.8 will decrease coverage by <.01%.

with the link to the coverage.io page is enough.

**Comment by stsewd** [comment]
@humitos what I saw, it just post one comment and then updates that comment on each commit, and yeah, it's very verbose, should be a way of a more simple report.

**Comment by stsewd** [comment]
Here we go https://codecov.io/gh/rtfd/readthedocs.org/ the request for the bot is sent, I guess someone with admin access needs to accept the invitation :man_shrugging:

**Comment by stsewd** [comment]
People at home can still see the reports, with `tox -e py36` and `tox -e coverage`

**Comment by ericholscher** [comment]
I poked around on codecov but didn't see what confirmation I needed. Is it to install the bot on GH? Seems like that is optional, so I'm not 100% sure what's required.

**Comment by ericholscher** [comment]
Ah, found the request in email -- approved. 👍

**Comment by stsewd** [comment]
I retriggered the job, looks like we need to set up a less verbose comment for the bot p:

> I'm a little worried that coverage will slow down our test runs

I didn't see any big difference locally, I think we are good.

**Comment by stsewd** [comment]
The bot is ready, @humitos are we ok with the bot's comment? If so, I'll merge this PR.

**Comment by humitos** [comment]
Haha! If it's not clear, I've trying to approve this PR --GitHub was working terrible :)

**Comment by ericholscher** [comment]
🎉

**Pull request merged**